inginious/frontend/pages/tasks.py

```diff
@@ -97,11 +98,11 @@ def GET(self, courseid, taskid, isLTI):
 
         self.user_manager.user_saw_task(username, courseid, taskid)
         
-        # Generate random inputs and save ot into db
+        # Generate random inputs and save it into db
         random_input_list = []
-        if username is not None:
-            random.seed(username + taskid + courseid)
-            random_input_list = [random.random() for i in range(task.get_number_input_random())]
+        random.seed(str(username if username is not None else "") + taskid + courseid + str(time.time() if task.regenerate_input_random() else ""))
```

Comment:
If I understand well, if the student opens the same task in two tabs, say Tab1 and Tab2, and then submit a correct answer in Tab1, it will fail because the random inputs have been regenerated. Maybe we should add a mechanism to, at least, refuse the submission for an outdated set of random inputs. What do you think ?

We can for instance simply return the generated set in the form and match with what was stored in DB.

Reply:
Yes, nice remark, I will fix this.